client/block-builder/src/lib.rs

```diff
@@ -153,11 +165,20 @@ where
 			})
 		} else {
 			self.api.map_api_result(|api| {
-				match api.apply_extrinsic_with_context(
```

Comment:
You need to bump the version of the api and make sure that `push_trusted` calls `apply_extrinsic` for `version < 5`.